docs/http/http-metrics.md

```diff
@@ -552,9 +552,22 @@ HTTP method names are case-sensitive and `http.request.method` attribute value M
 Instrumentations for specific web frameworks that consider HTTP methods to be case insensitive, SHOULD populate a canonical equivalent.
 Tracing instrumentations that do so, MUST also set `http.request.method_original` to the original value.
 
-**[2] `server.address`:** If an HTTP client request is explicitly made to an IP address, e.g. `http://x.x.x.x:8080`, then `server.address` SHOULD be the IP address `x.x.x.x`. A DNS lookup SHOULD NOT be used.
+**[2] `server.address`:** In HTTP/1.1, when the [request target](https://www.rfc-editor.org/rfc/rfc9112.html#name-request-target)
+is passed in its [absolute-form](https://www.rfc-editor.org/rfc/rfc9112.html#section-3.2.2),
+the `server.address` SHOULD match the host component of the request target.
 
-**[3] `server.port`:** When observed from the client side, and when communicating through an intermediary, `server.port` SHOULD represent the server port behind any intermediaries, for example proxies, if it's available.
+In all other cases, `server.address` SHOULD match the host component of the
+`Host` header in HTTP/1.1 or the `:authority` pseudo-header in HTTP/2 and HTTP/3.
+
+If an HTTP client request is explicitly made to an IP address, e.g. `http://x.x.x.x:8080`, then
```

Comment:
This wording goes against our goal with https://github.com/open-telemetry/semantic-conventions/issues/2443 and https://github.com/dotnet/runtime/issues/117049.

If a request goes to `http://x.x.x.x`, but the `Host` header is set, `x.x.x.x` will be only used to make the connection at transport level. `x.x.x.x` is absent at the HTTP level and `Host` is the _Server domain name_ users want to see in `server.address`.